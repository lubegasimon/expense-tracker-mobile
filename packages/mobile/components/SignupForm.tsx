import { Text, View, TextInput, StyleSheet } from "react-native";
import { useState } from "react";

export default function SignupForm() {
  const [username, setUsername] = useState("");

  return (
    <View style={styles.view}>
      <TextInput
        style={styles.input}
        onChangeText={setUsername}
        value={username}
        placeholder="Enter Your Username"
        placeholderTextColor="gray"
      />
    </View>
  );
}

const styles = StyleSheet.create({
  view: {
    flex: 1,
    justifyContent: "flex-start",
    alignItems: "center",
  },

  appName: {
    fontSize: 20,
    fontWeight: "bold",
  },
  input: {
    height: 40,
    margin: 12,
    borderColor: "gray",
    borderWidth: 1,
    padding: 10,
  },
});
