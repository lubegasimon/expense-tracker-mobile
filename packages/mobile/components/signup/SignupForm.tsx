import { Text, View, StyleSheet } from "react-native";

import UsernameField from "./UsernameField";
import EmailField from "./EmailField";
import PasswordField from "./PasswordField";
import ConfirmPasswordField from "./ConfirmPasswordField";

function SignupForm() {
  return (
    <View style={styles.view}>
      <Text> Create account </Text>
      <UsernameField />
      <EmailField />
      <PasswordField />
      <ConfirmPasswordField />
    </View>
  );
}

const styles = StyleSheet.create({
  view: {
    flex: 1,
    justifyContent: "flex-start",
    alignItems: "center",
    padding: 50,
  },

  appName: {
    fontSize: 20,
    fontWeight: "bold",
  },
});

export default SignupForm;
