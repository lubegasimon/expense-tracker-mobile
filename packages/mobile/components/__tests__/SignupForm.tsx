import { render, screen } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";

import SignupForm from "../signup/SignupForm";

describe("<SignupForm/>", () => {
  test("Signup form contains username text input", () => {
    render(
      <NavigationContainer>
        <SignupForm />
      </NavigationContainer>,
    );

    screen.getByPlaceholderText("Enter Your Username");
  });
});
