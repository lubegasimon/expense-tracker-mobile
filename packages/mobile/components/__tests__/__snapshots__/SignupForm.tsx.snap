// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupForm/> Signup form contains username text input 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
      "padding": 50,
    }
  }
>
  <Text>
     Create account 
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Your Username"
    placeholderTextColor="gray"
    style={
      {
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
        "width": 200,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Your Email"
    placeholderTextColor="gray"
    style={
      {
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
        "width": 200,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Your Password"
    placeholderTextColor="gray"
    style={
      {
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
        "width": 200,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Confirm Password"
    placeholderTextColor="gray"
    style={
      {
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
        "width": 200,
      }
    }
    value=""
  />
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#34b4eb",
          "borderColor": "white",
          "borderRadius": 10,
          "padding": 10,
          "width": 150,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Signup
      </Text>
    </View>
  </View>
</View>
`;
